modules/http4s/src/smithy4s/http4s/SimpleRestJsonBuilder.scala

```diff
@@ -19,12 +19,17 @@ package http4s
 
 import smithy4s.internals.InputOutput
 
-object SimpleRestJsonBuilder
+object SimpleRestJsonBuilder extends SimpleRestJsonBuilder(1024) {
+  def withMaxArity(maxArity: Int): SimpleRestJsonBuilder =
+    new SimpleRestJsonBuilder(maxArity) {}
+}
+abstract class SimpleRestJsonBuilder(maxArity: Int)
```

Comment:
minor: does it need to be abstract or what's the gain in the case?

Reply:
It doesnt .
I left it abstract for now to have a discussion, simply because SimpleProtocolBuilder is abstract and it doesn't need to be either.